src/clj/clj_diff/core.clj

```diff
@@ -17,27 +18,34 @@
 
   An index of -1 may appear in additions and is a special case which means to
   add the elements at the beginning of the sequence."
-  [a b]
-  (miller/diff a b))
+  [a b & [opts]]
+  (miller/diff a b opts))
 
 (defn- merge-patch
   [s edit-script delete-symbol]
-  (let [s (vec s)
+  (let [s         (fv/vec s)
         additions (:+ edit-script)
         deletions (:- edit-script)
-        s (reduce (fn [a b]
-                    (assoc a b delete-symbol))
-                  s
-                  deletions)
-        s (reduce (fn [a b]
-                    (let [index (first b)
-                          items (rest b)]
-                      (if (= index -1)
-                        (assoc a 0 (conj (vec items) (get a 0)))
-                        (assoc a index (conj items (get a index))))))
-                  s
-                  additions)]
-    (flatten s)))
+        s         (reduce (fn [a b]
+                            (assoc a b {::delete (get a b)}))
+                    s
+                    deletions)
+
+        s         (reduce (fn [a b]
+                            (let [index (first b)
+                                  items (fv/subvec b 1)]
+                              (if (= index -1)
+                                (assoc a 0 (fv/vector {::insert items} (get a 0)))
+                                (assoc a index (fv/vector (get a index) {::insert items})))))
+                    s
+                    additions)]
+    (mapcat
+      #(if (vector? %)
+         %
+         [%])
+      s)
```

Comment:
This might be a bit "bike shed"y, but I'd find this easier to read without binding temporary variables. If it was me I'd reach for `as->`, but I do like my arrows. Feel free to ignore if you'd rather not

Reply:
I appreciate the comment. I understand this comment regards the project as a whole and is not specific to this PR tho.

Reply:
Yep, it's a bit "procedural" in places. If you wanted to change it now, one could. But don't feel the need to if you don't think it's worth it